api/v1alpha1/contour_types.go

```diff
@@ -89,6 +96,49 @@ type NamespaceSpec struct {
 	RemoveOnDeletion bool `json:"removeOnDeletion,omitempty"`
 }
 
+// EndpointPublishing defines the schema to publish network endpoints and represents
+// the publishing type.
+type EndpointPublishing struct {
```

Comment:
In the future, fields will be added to support config knobs for the different endpoint publishing types.  `EndpointPublishing` will become a union type and `Type` will be the union discriminator.